collector/processor/coldstartprocessor/processor_test.go

```diff
@@ -0,0 +1,156 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coldstartprocessor // import "github.com/open-telemetry/opentelemetry-lambda/collector/processor/coldstartprocessor"
+
+import (
+	"context"
+	crand "crypto/rand"
+	"encoding/binary"
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+	semconv "go.opentelemetry.io/collector/semconv/v1.5.0"
+)
+
+type mockConsumer struct {
+	consumed int
+}
+
+func (c *mockConsumer) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	c.consumed += td.SpanCount()
+	return nil
+}
+
+func (c *mockConsumer) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: true}
+}
+
+func addExecutionSpan(span ptrace.Span) {
+	var rngSeed int64
+	_ = binary.Read(crand.Reader, binary.LittleEndian, &rngSeed)
+	randSource := rand.New(rand.NewSource(rngSeed))
+	tid := pcommon.TraceID{}
+	_, _ = randSource.Read(tid[:])
+	span.SetTraceID(tid)
+	span.Attributes().PutStr(semconv.AttributeFaaSExecution, "af9d5aa4-a685-4c5f-a22b-444f80b3cc28")
+}
+
+func TestProcessor(t *testing.T) {
+	testCases := []struct {
+		desc          string
+		input         ptrace.Traces
+		expected      ptrace.Traces
+		expectedError error
+	}{
+		{
+			desc:          "no traces",
+			input:         ptrace.NewTraces(),
+			expected:      ptrace.NewTraces(),
+			expectedError: processorhelper.ErrSkipProcessingData,
+		},
+		{
+			desc: "coldstart without executioin",
+			input: func() ptrace.Traces {
+				td := ptrace.NewTraces()
+				span := td.ResourceSpans().AppendEmpty().ScopeSpans().AppendEmpty().Spans().AppendEmpty()
+				span.Attributes().PutBool(semconv.AttributeFaaSColdstart, true)
+				return td
+			}(),
+			expected: func() ptrace.Traces {
+				td := ptrace.NewTraces()
+				td.ResourceSpans().AppendEmpty()
+				td.ResourceSpans().RemoveIf(func(rs ptrace.ResourceSpans) bool { return true })
+				return td
+			}(),
+			expectedError: processorhelper.ErrSkipProcessingData,
+		},
+		{
+			desc: "execution without coldstart",
+			input: func() ptrace.Traces {
+				td := ptrace.NewTraces()
+				addExecutionSpan(td.ResourceSpans().AppendEmpty().ScopeSpans().AppendEmpty().Spans().AppendEmpty())
+				return td
+			}(),
+			expected: func() ptrace.Traces {
+				td := ptrace.NewTraces()
+				addExecutionSpan(td.ResourceSpans().AppendEmpty().ScopeSpans().AppendEmpty().Spans().AppendEmpty())
+				return td
+			}(),
+		},
+		{
+			desc: "faas.execution and faas.coldstart with coldstart is first",
+			input: func() ptrace.Traces {
+				td := ptrace.NewTraces()
+				span := td.ResourceSpans().AppendEmpty().ScopeSpans().AppendEmpty().Spans().AppendEmpty()
```

Comment:
What about non-empty resource or scope attributes?  Or if they're different between the execution span and the coldstart span?  I think with these tests the implementation could drop all resource attributes from every span it processes and they'd still pass.

Reply:
good catch, i added some tests to validate the resource and scope attributes. the current implementation gives preference to the faas execution span's resource+scope details over the coldstart, as it should contain more meaningful data. The coldstart span's timing is the key information.